client/state/selectors/get-network-sites.js

```diff
@@ -14,23 +14,23 @@ import { isMainSiteOf } from 'state/selectors';
 import { getSite, isJetpackSiteMainNetworkSite } from 'state/sites/selectors';
 
 /**
- * Returns the secondary sites of a given site
- * Returns null if main site is not found or if the site is not a main site
+ * Returns the member sites (main or secondary) of given network site
+ * Returns null if main site is not found or if the site is not a main site of a network
  *
- * @param  {Object}  state     Global state tree
- * @param  {Number}  siteId    The ID of the main site for which we're retrieving the secondary sites
- * @return {?Array}            Array of secondary sites
+ * @param  {Object}  state       Global state tree
+ * @param  {Number}  mainSiteId  The ID of the main site for which we're retrieving the network sites
+ * @return {?Array}              Array of network sites (the main one and the secondary ones)
  */
 export default createSelector(
-	( state, siteId ) => {
-		const isMainNetworkSite = isJetpackSiteMainNetworkSite( state, siteId );
-		if ( ! isMainNetworkSite ) {
+	( state, mainSiteId ) => {
+		if ( ! isJetpackSiteMainNetworkSite( state, mainSiteId ) ) {
 			return null;
 		}
-		const siteIds = Object.keys( get( state, 'sites.items', {} ) );
-		return siteIds
-			.filter( secondarySiteId => isMainSiteOf( state, siteId, secondarySiteId ) )
-			.map( secondarySiteId => getSite( state, secondarySiteId ) );
+
+		return filter(
+			state.sites.items,
```

Comment:
Are `state.sites` always defined? Wouldn't it be safer using `get` here?

Reply:
Yes, `state.sites.items` is always defined. And if you look into the dependents of this selector, they also do a direct access to `state.sites.items`. If it wasn't defined, this selector in its current form would be throwing exceptions anyway.

I'd reserve `get` only for cases where there's some plausible reason why some member of the property chain shouldn't exist. I'm not very fond of using it by default, "just in case".